src/lib.rs

```diff
@@ -267,77 +302,184 @@ fn interface_and_mtu_windows(socket: &UdpSocket) -> Result<(String, usize), Erro
 #[doc(hidden)]
 #[deprecated(since = "0.1.2", note = "Use `interface_and_mtu()` instead")]
 pub fn interface_mtu(remote: &SocketAddr) -> Result<usize, Error> {
-    interface_and_mtu(remote).map(|(_, mtu)| mtu)
+    interface_and_mtu(SocketAddrs::Remote(*remote)).map(|(_, mtu)| mtu)
 }
 
 #[doc(hidden)]
 #[deprecated(since = "0.1.2", note = "Use `interface_and_mtu()` instead")]
 pub fn get_interface_mtu(remote: &SocketAddr) -> Result<usize, Error> {
-    interface_and_mtu(remote).map(|(_, mtu)| mtu)
+    interface_and_mtu(SocketAddrs::Remote(*remote)).map(|(_, mtu)| mtu)
 }
 
 #[cfg(test)]
 mod test {
-    use std::net::ToSocketAddrs;
-
-    use log::warn;
-
-    fn check_mtu(sockaddr: &str, ipv4: bool, expected: usize) {
-        let addr = sockaddr
-            .to_socket_addrs()
-            .unwrap()
-            .find(|a| a.is_ipv4() == ipv4);
-        if let Some(addr) = addr {
-            match super::interface_and_mtu(&addr) {
-                Ok((_, mtu)) => assert_eq!(mtu, expected),
-                Err(e) => {
-                    // Some GitHub runners don't have IPv6. Just warn if we can't get the MTU.
-                    assert!(addr.is_ipv6());
-                    warn!("Error getting MTU for {sockaddr}: {e}");
+    use std::{
+        env,
+        net::{IpAddr, Ipv4Addr, Ipv6Addr, SocketAddr, UdpSocket},
+    };
+
+    use rand::Rng;
+
+    use crate::interface_and_mtu;
+
+    #[derive(Debug)]
+    struct NameMtu<'a>(Option<&'a str>, usize);
+
+    impl PartialEq<NameMtu<'_>> for (String, usize) {
+        fn eq(&self, other: &NameMtu<'_>) -> bool {
+            other.0.map_or(true, |name| name == self.0) && other.1 == self.1
+        }
+    }
+
+    #[cfg(target_os = "macos")]
+    const LOOPBACK: NameMtu = NameMtu(Some("lo0"), 16_384);
+    #[cfg(target_os = "linux")]
+    const LOOPBACK: NameMtu = NameMtu(Some("lo"), 65_536);
+    #[cfg(target_os = "windows")]
+    const LOOPBACK: NameMtu = NameMtu(Some("loopback_0"), 4_294_967_295);
+
+    // Non-loopback interface names are unpredictable, so we only check the MTU.
+    #[cfg(target_os = "macos")]
+    const INET: NameMtu = NameMtu(None, 1_500);
+    #[cfg(target_os = "linux")]
+    const INET: NameMtu = NameMtu(None, 1_500);
+    #[cfg(target_os = "windows")]
+    const INET: NameMtu = NameMtu(None, 1_500);
+
+    //  The tests can run in parallel, so try and find unused ports for all the tests.
+    fn socket_with_addr(local_ip: IpAddr) -> SocketAddr {
+        loop {
+            let port = rand::thread_rng().gen_range(1024..65535);
+            let socket = UdpSocket::bind(SocketAddr::new(local_ip, port));
+            match socket {
+                Ok(socket) => {
+                    if let Ok(saddr) = socket.local_addr() {
+                        // We found an unused port.
+                        return saddr;
+                    }
```

Comment:
I agree with codecov about these uncovered else clauses.  Maybe you can use `.unwrap()` instead.  My understanding is that the code coverage won't complain about that.